src/lib/BasicSideMenuItem.tsx

```diff
@@ -0,0 +1,86 @@
+import { Divider, List, ListItem, ListItemButton, ListItemIcon, ListItemText } from "@mui/material"
+
+import {
+    ChatBubble as ChatBubbleIcon,
+    FactoryOutlined,
+    KeyOutlined,
+    AssignmentTurnedInOutlined,
+    TextSnippetOutlined,
+    HighlightAlt,
+    SupervisorAccount as AccountIcon,
+    Map as MapIcon,
+    Tablet as TabletIcon,
+    Update as UpdateIcon,
+    Security as AdminIcon,
+    ContentPaste as ClipBoardIcon,
+    FitScreen,
+    Event as EventIcon,
+    Logout as LogoutIcon
+} from '@mui/icons-material'
+
+const iconmap: any = {
+    factory: FactoryOutlined,
+    key: KeyOutlined,
+    done: AssignmentTurnedInOutlined,
+    docs: TextSnippetOutlined,
+    hightlight: HighlightAlt,
+    map: MapIcon,
+    account: AccountIcon,
+    tablet: TabletIcon,
+    update: UpdateIcon,
+    admin: AdminIcon,
+    clipboard: ClipBoardIcon,
+    fitscreen: FitScreen,
+    chatBubble: ChatBubbleIcon,
+    event: EventIcon,
+    logout: LogoutIcon
+}
+
+function makeIcon(name: string) {
+    const Icon = iconmap[name]
+    return <Icon />
+}
+
+// TODO: Grab the role from the redux store
+const userRole = 'admin'
+
+// TODO: move to utils
+// TODO: allow custom authorization function
+function isAuthorized(userRole: string, authorizedRoles: any): boolean {
+    return authorizedRoles.hasOwnProperty(userRole) && authorizedRoles[userRole] === true
+}
+
+function BasicSideMenuItem(props: any) {
+    const { spec, sectionKey, onItemSelect } = props
```

Comment:
The issue I'm running into is Gubu not failing on Child({}), for example 
```
item: Child({
   kind: Number,
   label: String,
   icon: String,
   path: String,
   access: Child(Boolean, {})
})
```
This should fail because clearly `kind` shouldn't be a `Number`. What am I doing wrong?